### DISCUSSION TITLE
Feature Request: Enhanced User Blocking to Restrict Profile Access

### DISCUSSION THREAD

**Discussion by haleygelato** [community/community#80968] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Hi! I noticed that even after blocking someone on GitHub, they can still check out my profile. Wish there was a feature where if you block someone, they're completely unable to view your profile. TY!

**Comment by j0rd1s3rr4n0** [comment]
Hi @haleygelato,

Thanks for your suggestion. I understand that the blocking feature could be useful to prevent others from making Pull Requests. Since you can set both your repositories and profile to public or private, is that the main reason behind this request?

**Reply by leeringrose** [reply]
I think she wants to hide his profile from a specific person. That's what she's asking. If I am right, @haleygelato ?

**Reply by haleygelato** [reply]
Yes, you're correct! I'm looking to hide my profile from specific individuals, similar to how blocking works on Facebook. This feature would be beneficial for maintaining privacy and controlling who can see personal details like location, company, and other profile information. 

When I set my profile to private, information such as bio, location, and company still remains public. Also, it's not very convenient since I want to block only certain people while still displaying these details to others. 

Thank you!

**Comment by bydevmar** [comment]
Hi! Blocking on GitHub currently restricts interaction, but not profile visibility. Your feedback is valuable, and it's worth considering for platform improvements. Thanks for sharing your thoughts!
